In [3]:
import pandas as pd

https://www.kaggle.com/datasets/sohaibdevv/consumer-shopping-behavior-and-preference-study-2026

In [4]:
compras = pd.read_csv('compras_online_store.csv')

In [5]:
compras.shape

(11789, 25)

In [6]:
compras.head()

,age,monthly_income,daily_internet_hours,smartphone_usage_years,social_media_hours,online_payment_trust_score,tech_savvy_score,monthly_online_orders,monthly_store_visits,avg_online_spend,...,free_return_importance,product_availability_online,impulse_buying_score,need_touch_feel_score,brand_loyalty_score,environmental_awareness,time_pressure_level,gender,city_tier,shopping_preference
0,56,221111,6.5,12,0.7,1,6,16,16,28551,...,7,7,1,3,6,5,2,Other,Tier 3,Store
1,69,96029,8.2,13,2.7,6,9,14,1,124056,...,3,4,9,6,8,1,7,Male,Tier 3,Hybrid
2,46,19055,6.4,4,2.1,10,8,2,0,81939,...,4,10,1,1,3,3,3,Female,Tier 3,Store
3,32,53170,6.4,11,0.7,2,10,20,3,35901,...,10,2,4,8,2,6,6,Female,Tier 1,Store
4,60,244016,6.0,5,0.7,2,5,18,16,131971,...,2,5,8,9,7,1,6,Male,Tier 3,Store


In [134]:
compras['city_tier'] = compras['city_tier'].str.replace('Tier ', '')
compras.head()

,age,monthly_income,daily_internet_hours,smartphone_usage_years,social_media_hours,online_payment_trust_score,tech_savvy_score,monthly_online_orders,monthly_store_visits,avg_online_spend,...,free_return_importance,product_availability_online,impulse_buying_score,need_touch_feel_score,brand_loyalty_score,environmental_awareness,time_pressure_level,gender,city_tier,shopping_preference
0,56,221111,6.5,12,0.7,1,6,16,16,28551,...,7,7,1,3,6,5,2,Other,3,Store
1,69,96029,8.2,13,2.7,6,9,14,1,124056,...,3,4,9,6,8,1,7,Male,3,Hybrid
2,46,19055,6.4,4,2.1,10,8,2,0,81939,...,4,10,1,1,3,3,3,Female,3,Store
3,32,53170,6.4,11,0.7,2,10,20,3,35901,...,10,2,4,8,2,6,6,Female,1,Store
4,60,244016,6.0,5,0.7,2,5,18,16,131971,...,2,5,8,9,7,1,6,Male,3,Store


O arquivo trata-se de um conjunto de informações sobre compras online e em lojas fisicas, para um estudo comprativo entre as duas opções de compras.
Ele possui 11789 linhas e 25 colunas.
As colunas são: idade, renda mensal, horas diárias na internet, anos de uso do smartphone, horas nas redes sociais, nivel confiança em pagamento online, nível conhecimento tecnológico, pedidos online mensais, visitas lojas mensais, gasto médio online, importância devolução grátis, disponibilidade produto online, nível compra por impulso, nível necessidade de tocar sentir(avaliar fisicamente), nível lealdade marca, conscientização ambiental, nivel pressão tempo, genero, porte da cidade, preferências compras.
O data set não possui dados nulos.
Na coluna 'city_tier' foi excluído o termo 'Tier" que se repetia em todas as linhas.
O dataset apresentada as seguintes variáveis: Interger, Float, Strings.
Das 25 colunas, 20 são do tipo int, 2 flot e 3 string.


Perguntas:
1. O nível de consciencia ambiental aumenta ou diminui de acordo com a idade e renda?
2. A quantidade de anos de uso do smartphone influencia na quantidade de compras online?
3. Quem passa mais horas nas redes sociais tem um gasto maior em lojas fisicas ou online?
4. o nível de cohecimento tecnológico interfere no nível de confiança nos meios de pagamento online?.
5. Qual a média de idade das pessoas que tem o maior volume de compras online?
6. Qual a preferencia de canal de compra de acordo com o porte da cidade
7. Qual a preferencia de canal de compras de acordo com o  genero?
8. Quem tem maior nível de compras por impulso prefere comprar online ou fisicamente
9. Relação da renda com o canal preferido de compras
10. Qual a média de idade das pessoas que tem o maior volume de compras fisicas?


In [138]:
compras.describe()

,age,monthly_income,daily_internet_hours,smartphone_usage_years,social_media_hours,online_payment_trust_score,tech_savvy_score,monthly_online_orders,monthly_store_visits,avg_online_spend,...,return_frequency,avg_delivery_days,delivery_fee_sensitivity,free_return_importance,product_availability_online,impulse_buying_score,need_touch_feel_score,brand_loyalty_score,environmental_awareness,time_pressure_level
count,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000,...,11789.00000,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000,11789.000000
mean,48.729409,131704.282382,6.011367,7.597930,2.514471,5.498770,5.534312,24.677581,9.482144,74554.929341,...,4.46747,3.999661,5.468827,5.462041,5.518704,5.486386,5.485368,5.532021,5.448554,5.504114
std,17.899445,68120.726684,1.976811,4.011628,1.263047,2.880366,2.887251,14.431277,5.728825,43167.126595,...,2.88545,1.995881,2.870195,2.882177,2.867613,2.877918,2.877264,2.848796,2.872740,2.876561
min,18.000000,15005.000000,1.000000,1.000000,0.000000,1.000000,1.000000,0.000000,0.000000,523.000000,...,0.00000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000
25%,33.000000,72450.000000,4.600000,4.000000,1.600000,3.000000,3.000000,12.000000,5.000000,36797.000000,...,2.00000,2.000000,3.000000,3.000000,3.000000,3.000000,3.000000,3.000000,3.000000,3.000000
50%,49.000000,131916.000000,6.000000,8.000000,2.500000,5.000000,6.000000,25.000000,9.000000,74859.000000,...,4.00000,4.000000,5.000000,5.000000,6.000000,5.000000,5.000000,6.000000,5.000000,6.000000
75%,64.000000,190505.000000,7.400000,11.000000,3.400000,8.000000,8.000000,37.000000,14.000000,112134.000000,...,7.00000,6.000000,8.000000,8.000000,8.000000,8.000000,8.000000,8.000000,8.000000,8.000000
max,79.000000,249989.000000,12.000000,14.000000,6.000000,10.000000,10.000000,49.000000,19.000000,149996.000000,...,9.00000,7.000000,10.000000,10.000000,10.000000,10.000000,10.000000,10.000000,10.000000,10.000000


In [137]:
compras.info()

<class 'pandas.DataFrame'>
RangeIndex: 11789 entries, 0 to 11788
Data columns (total 25 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   age                          11789 non-null  int64  
 1   monthly_income               11789 non-null  int64  
 2   daily_internet_hours         11789 non-null  float64
 3   smartphone_usage_years       11789 non-null  int64  
 4   social_media_hours           11789 non-null  float64
 5   online_payment_trust_score   11789 non-null  int64  
 6   tech_savvy_score             11789 non-null  int64  
 7   monthly_online_orders        11789 non-null  int64  
 8   monthly_store_visits         11789 non-null  int64  
 9   avg_online_spend             11789 non-null  int64  
 10  avg_store_spend              11789 non-null  int64  
 11  discount_sensitivity         11789 non-null  int64  
 12  return_frequency             11789 non-null  int64  
 13  avg_delivery_days          

In [140]:
# 1. O nível de consciencia ambiental aumenta ou diminui de acordo com a idade e renda?

#verificando a faixa etária do dataset
compras['age'].sort_values().unique()

#definindo 3 faixas de idade 
# Faixa 1 - 18 a 35 anos
# Faixa 2 - 36 a 55 anos
# Faixa 3 - 56 a 79 anos
faixa1 = compras['age'] <= 35
faixa2 = (compras['age'] >= 36) & (compras['age'] <= 55)
faixa3 = (compras['age'] >= 56) & (compras['age'] <= 79)

#Calculando a media do nível de consciencia ambiental por faixa etária
media_consc_ambiental_f1 = compras[faixa1]['environmental_awareness'].mean().round(2)
media_consc_ambiental_f2 = compras[faixa2]['environmental_awareness'].mean().round(2)
media_consc_ambiental_f3 = compras[faixa3]['environmental_awareness'].mean().round(2)
print(f'Média de nível de consciencia ambiental para pessoas de baixa renda é {media_consc_ambiental_f1}')
print(f'Média de nível de consciencia ambiental para pessoas de baixa renda é {media_consc_ambiental_f2}')
print(f'Média de nível de consciencia ambiental para pessoas de baixa renda é {media_consc_ambiental_f3}')

# Verificando a faixa de renda do dataset
compras['monthly_income'].describe()

# Definindo as 3 faixas de renda
baixa_renda = compras['monthly_income'] <= 72450
media_renda = (compras['monthly_income'] > 72450) & (compras['monthly_income'] <= 190505)
alta_renda = compras['monthly_income'] > 190505

# Calculando a media do nível de consciencia ambiental por faixa etária
media_baixa_renda = compras[baixa_renda]['environmental_awareness'].mean().round(2)
media_media_renda = compras[media_renda]['environmental_awareness'].mean().round(2)
media_alta_renda = compras[alta_renda]['environmental_awareness'].mean().round(2)
print('=-' * 35)
print(f'Média de nível de consciencia ambiental para pessoas de baixa renda é {media_baixa_renda}')
print(f'Média de nível de consciencia ambiental para pessoas de baixa renda é {media_media_renda}')
print(f'Média de nível de consciencia ambiental para pessoas de baixa renda é {media_alta_renda}')



Média de nível de consciencia ambiental para pessoas de baixa renda é 5.43
Média de nível de consciencia ambiental para pessoas de baixa renda é 5.45
Média de nível de consciencia ambiental para pessoas de baixa renda é 5.46
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
Média de nível de consciencia ambiental para pessoas de baixa renda é 5.41
Média de nível de consciencia ambiental para pessoas de baixa renda é 5.46
Média de nível de consciencia ambiental para pessoas de baixa renda é 5.47


Resposta: Conforme verificamos, não há variação considerável no nível de consciencia ambiental nas
       diferentes faixas de renda e de idade.

In [141]:
# 2. A quantidade de anos de uso do smartphone influencia na quantidade de compras online?


# Agrupando por anos de uso e calculando a média de gastos para cada ano
media_gasto_online_store = compras.groupby('smartphone_usage_years')[['avg_online_spend', 'avg_store_spend']].mean().round(2).reset_index()


# Verificando os resultados
display(media_gasto_online_store.sort_index())



,smartphone_usage_years,avg_online_spend,avg_store_spend
0,1,74927.31,75130.44
1,2,73998.91,75610.45
2,3,73907.46,76010.04
3,4,75047.72,76716.21
4,5,73161.54,72695.50
5,6,76866.49,75792.82
6,7,72797.79,73899.06
7,8,73384.06,76573.45
8,9,74716.19,74888.36
9,10,76274.83,74767.27


Resposta: Olhando para a média de gastos, seja online ou em lojas físicas, vemos que a quantidade de anos de uso de smartphone não tem influência.

In [142]:
# 3. Quem passa mais horas nas redes sociais tem um gasto maior em lojas fisicas ou online?

'''Nesta consulta, foi necessário fazer o arredontamento das horas, pois, sem isso, a coluna "daily_internet_hours" 
ficaria muito extensa e prejucaria a visualização dos dados'''

m_gastos_online_store = compras.groupby(compras ['daily_internet_hours'].round().astype(int))[['avg_online_spend', 'avg_store_spend']].mean().round(2).reset_index()


display(m_gastos_online_store)




,daily_internet_hours,avg_online_spend,avg_store_spend
0,1,75657.52,73646.78
1,2,73234.17,77062.15
2,3,74068.15,76045.43
3,4,73574.03,76154.61
4,5,76169.33,75163.53
5,6,73998.62,76392.49
6,7,73666.53,74518.17
7,8,75362.46,75611.87
8,9,75552.11,77483.63
9,10,74335.88,73009.90


Resposta: Não foi verificada relação entre o tempo que a pessoa navega na internet e o volume de gastos, seja online ou em lojas físicas.

In [143]:
# 4. o nível de cohecimento tecnológico interfere no nível de confiança nos meios de pagamento online?.

tecnol_pgto_digital = (compras.groupby('tech_savvy_score')['online_payment_trust_score'].mean().round(2).reset_index()
.rename(columns={'tech_savvy_score': 'Nível de Conhecimento Tecnológico',
                 'online_payment_trust_score': 'Média de Confiança no Pagamento'}))

display(tecnol_pgto_digital)




,Nível de Conhecimento Tecnológico,Média de Confiança no Pagamento
0,1,5.58
1,2,5.54
2,3,5.54
3,4,5.65
4,5,5.52
5,6,5.38
6,7,5.42
7,8,5.52
8,9,5.39
9,10,5.45


Resposta: Analisando os diferentes níveis de conhecimento tecnológico, vemos que não há interferência na confiança nos meios de pagamento digital

In [144]:
# 5. Qual a média de idade das pessoas que tem o maior volume de compras online?


# Definindo 5 faixas de idade 
#Faixa1 = 'age' <= 27 
#Faixa2 = 'age' >= 28 & 'age' <= 37 
#Faixa3 = 'age' >= 38 & 'age' <= 47  
#Faixa4 = 'age' >= 48 & 'age' <= 57 
#Faixa5 = 'age' >= 58 & 'age' <= 67 
#Faixa6 = 'age' >= 68

m_idade_x_comprasf1 = compras[compras['age']<= 27]['avg_online_spend'].mean().round(2)
m_idade_x_comprasf2 = compras[(compras['age']>= 28) & (compras['age']<= 37)]['avg_online_spend'].mean().round(2)
m_idade_x_comprasf3 = compras[(compras['age']>= 38) & (compras['age']<= 47)]['avg_online_spend'].mean().round(2)
m_idade_x_comprasf4 = compras[(compras['age']>= 48) & (compras['age']<= 57)]['avg_online_spend'].mean().round(2)
m_idade_x_comprasf5 = compras[(compras['age']>= 58) & (compras['age']<= 67)]['avg_online_spend'].mean().round(2)
m_idade_x_comprasf6 = compras[(compras['age']>= 68)]['avg_online_spend'].mean().round(2)

media_por_faixa_etaria = [['18 a 27', m_idade_x_comprasf1], ['28 a 37', m_idade_x_comprasf2],
['38 a 47', m_idade_x_comprasf3],['48 a 57', m_idade_x_comprasf4],['58 a 67', m_idade_x_comprasf5],
['67 a 79', m_idade_x_comprasf6]]


df_gasto_faixa_etaria = pd.DataFrame(media_por_faixa_etaria, columns=['Faixa Etária', 'Média compras Online'])
display(df_gasto_faixa_etaria)



,Faixa Etária,Média compras Online
0,18 a 27,72971.86
1,28 a 37,74628.04
2,38 a 47,74287.63
3,48 a 57,74164.52
4,58 a 67,75348.44
5,67 a 79,75629.51


Resposta: Com base nas informações da media de gastos com compras online por faixa etária, percebe-se um discreto aumento nas faixas acima de 58 anos.

In [18]:
# 6. Qual a preferencia de canal de compra de acordo com o porte da cidade

porte_cidade_pref_canal = (compras.groupby('city_tier')['shopping_preference'].value_counts().reset_index()
.rename(columns={'city_tier': 'Nível da Cidade', 'shopping_preference':'Canal Compras Favorito', 'count': 'Quant Pessoas'}))                           

display(porte_cidade_pref_canal)


,Nível da Cidade,Canal Compras Favorito,Quant Pessoas
0,Tier 1,Store,3444
1,Tier 1,Online,400
2,Tier 1,Hybrid,138
3,Tier 2,Store,3371
4,Tier 2,Online,378
5,Tier 2,Hybrid,109
6,Tier 3,Store,3429
7,Tier 3,Online,398
8,Tier 3,Hybrid,122


In [163]:
cid_nivel1 = (compras[compras['city_tier'] == '1']['shopping_preference'].value_counts().reset_index()
              .rename(columns ={'shopping_preference': 'Canal compras favorito', 'count': 'Quantidade'}))

cid_nivel2 = (compras[compras['city_tier'] == '2']['shopping_preference'].value_counts().reset_index()
              .rename(columns ={'shopping_preference': 'Canal compras favorito', 'count': 'Quantidade'}))

cid_nivel3 = (compras[compras['city_tier'] == '3']['shopping_preference'].value_counts().reset_index()
              .rename(columns ={'shopping_preference': 'Canal compras favorito', 'count': 'Quantidade'}))

display(cid_nivel1)
display(cid_nivel2)
display(cid_nivel3)

,Canal compras favorito,Quantidade
0,Store,3444
1,Online,400
2,Hybrid,138


,Canal compras favorito,Quantidade
0,Store,3371
1,Online,378
2,Hybrid,109


,Canal compras favorito,Quantidade
0,Store,3429
1,Online,398
2,Hybrid,122


Resposta: Em todos os níveis de cidade, a preferência por compras em lojas físicas prevalece sobre os demais modos de compra.

In [179]:
# 7. Qual a preferencia de canal de compras de acordo com o  genero?

gen_canal_favor = (compras.groupby('gender')['shopping_preference'].value_counts().reset_index()
                   .rename(columns={'gender': 'Gênero', 'shopping_preference': 'Preferência compras', 'count': 'Quantidade'}))

display(gen_canal_favor)

,Gênero,Preferência compras,Quantidade
0,Female,Store,3378
1,Female,Online,408
2,Female,Hybrid,145
3,Male,Store,3486
4,Male,Online,378
5,Male,Hybrid,102
6,Other,Store,3380
7,Other,Online,390
8,Other,Hybrid,122


Resposta: Em todos os gêneros, o canal favorito para compras são as lojas físicas.

In [17]:
# 8. Quem tem maior nível de compras por impulso por genero

compras_impulso_alto = (compras[compras['impulse_buying_score'] >= 7].groupby('gender')
['impulse_buying_score'].count().reset_index().rename(columns={'gender':'Gênero', 'impulse_buying_score':'Quant Pessoas Alto Nível Compras por Impulso'}))
display(compras_impulso_alto)


,Gênero,Quant Pessoas Alto Nível Compras por Impulso
0,Female,1571
1,Male,1563
2,Other,1574


Resposta: Observando os dados dos níveis mais altos de compras por impulso (igual e acima de 7) observamos que não há difenrença significativa entre os gêneros.

In [14]:
# 9. Relação da renda com o canal preferido de compras

renda_desconto = (compras.groupby('discount_sensitivity')['monthly_income'].agg(['count', 'mean']).round(2)
                  .reset_index().rename(columns={'discount_sensitivity':'Nível sensibiidade a descontos', 
                 'count':'Quant pessoas por nível','mean':'Média renda por nível'}))
display(renda_desconto)


,Nível sensibiidade a descontos,Quant pessoas por nível,Média renda por nível
0,1,1207,130222.03
1,2,1172,132757.44
2,3,1146,133904.16
3,4,1153,128573.66
4,5,1204,133863.50
5,6,1203,130512.24
6,7,1167,133067.95
7,8,1216,131430.65
8,9,1116,132291.00
9,10,1205,130512.74


Resposta: Conforme tabela, não se verifica maior ou menor sensibilidade á descontos com puxar a quantidade de pessoas

In [22]:
# 10. Qual a média de idade das pessoas que tem o maior volume de compras fisicas?